src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -25,6 +26,32 @@ case class Task(
     */
   lazy val marathonTask: MarathonTask = TaskSerializer.marathonTask(this)
 
+  lazy val launchedMesosId: Option[MesosProtos.TaskID] = launched.map { _ =>
+    // it doesn't make sense for an unlaunched task
+
+    var id = taskId.idString
+    if (launchCounter > 0) {
```

Comment:
Kind of implicit - launchCounter>0 == resident task?
Perhaps create a separate def and use it to make it more explicit?
```scala 
def isResident = launchCounter>0
````